src/engine/engineworkerscheduler.h

```diff
@@ -30,7 +31,7 @@ class EngineWorkerScheduler : public QThread {
     // runWorkers was run. This should only be touched from the engine callback.
     bool m_bWakeScheduler;
 
-    FIFO<EngineWorker*> m_scheduleFIFO;
+    QHash<EngineWorker*, bool> m_workerStates;
```

Comment:
How do we ensure that QHash is never modified from another thread while being read/iterated from the scheduler thread? This should be documented in every detail, because we are in multi-threaded land here! I don't recognize it at first sight.